x-pack/test/reporting_api_integration/services.ts

```diff
@@ -139,21 +138,12 @@ export function ReportingAPIProvider({ getService }: FtrProviderContext) {
       log.debug('ReportingAPI.deleteAllReports');
 
       // ignores 409 errs and keeps retrying
-      const deleted$ = Rx.interval(100).pipe(
-        switchMap(() =>
-          esSupertest
-            .post('/.reporting*/_delete_by_query')
-            .send({ query: { match_all: {} } })
-            .then(({ status }) => status)
-        ),
-        filter((status) => status === 200),
-        mapTo(true),
-        first(),
-        timeout(5000)
-      );
-
-      const reportsDeleted = await deleted$.toPromise();
-      expect(reportsDeleted).to.be(true);
+      await retry.tryForTime(5000, async () => {
```

Comment:
Curious to see how this handles the 409's we return sometimes. I'm not super familiar with how `retry.tryForTime` works internally

Reply:
The `.expect(200)` will throw an error for any response code different than `200` and this will be caught by the retry, that will then execute the code block again (until the timeout is reached).